### DISCUSSION TITLE
Error: array_merge(): Argument #2 must be of type array, bool given

### DISCUSSION THREAD

**Discussion by hafifi** [opcodesio/log-viewer#259] [Q&A] [answered]
I get this error when accessing the log viewer from production server:
`array_merge(): Argument #2 must be of type array, bool given {"exception":"[object] (TypeError(code: 0): array_merge(): Argument #2 must be of type array, bool given at /home/path-to-laravel-app/vendor/opcodesio/log-viewer/src/LogViewerService.php:58)`

<img width="1021" alt="Screenshot 2023-08-16 at 11 01 42 PM" src="https://github.com/opcodesio/log-viewer/assets/1578426/a3042851-ef63-4023-b9a3-547be2163db2">

I can access from my local development machine. Any ideas what might goes wrong? I removed all logs and the error still persist. I'm on PHP8.1

**Comment by hafifi** [comment, marked as answer]
I think I found the problem, some of the included files path in config file might done something. I removed all other included files and only keep the laravel log and it works.
Change from this:
```
    'include_files' => [
        '*.log',
        '**/*.log',

        // You can include paths to other log types as well, such as apache, nginx, and more.
        '/var/log/httpd/*',
        '/var/log/nginx/*',

        // MacOS Apple Silicon logs
        '/opt/homebrew/var/log/nginx/*',
        '/opt/homebrew/var/log/httpd/*',
        '/opt/homebrew/var/log/php-fpm.log',
        '/opt/homebrew/var/log/postgres*log',
        '/opt/homebrew/var/log/redis*log',
        '/opt/homebrew/var/log/supervisor*log',

        // '/absolute/paths/supported',
    ],
```

To just this:
```
'include_files' => [
        '*.log',
        '**/*.log',
    ],
```

And walla!

**Reply by arukompas** [reply]
@hafifi hey! Log Viewer **v3.0.2** should also handle this correctly and continue, even if it can't access certain folders/files configured.

**Reply by hafifi** [reply]
Thank you!

**Comment by arukompas** [comment]
This is useful, thanks @hafifi !

Is there a chance that the /var/log/* files were inaccessible to the web process? That's something I'm also trying to work out, but I haven't seen that error before.

Either way, this is really helpful, thanks for posting 😃

**Reply by hafifi** [reply]
The production server is running with [runcloud](https://runcloud.io), so it might be some permissions configured is different from standard linux box (my staging linux environment has no issue as well, just with runcloud managed server). I'll do some testing and update if something came up :)
